lib/src/path_parser.cc

```diff
@@ -143,14 +143,18 @@ namespace hocon {
             return token_list { t };
         }
 
+        // The split iterator tacks an empty string onto the end of the iterator if the string
+        // ends in the split character; so we need to remove trailing '.'
+        string text_to_split = boost::trim_right_copy_if(token_text, is_any_of("."));
```

Comment:
So these address two separate issues: line 165 handles the case where token_text is e.g. "a.b.c", in which c is considered a key to a final value, I believe (not 100% sure, but that's kind of what the tests look like). My problem was in the case of "a.b.c.", where the split iterator returned an extra empty string at the end, resulting in double "." when using the tokens to make a path.